### PR TITLE
Use flexbox on app page

### PR DESCRIPTION
**Pull request by rullzer**
The amount of white space on the app page was annoying me. So I decided to give flexbox a try.

Before:
before

After:
after

Which looks a lot better to me.

However it does need work as you can see on the notenabled page:
not_enabled

Far from done.
- [x] Experimental apps display
- [ ] Fix per app layout a bit more
- [ ] Full Safari/IE support
- [ ] Restrict to apps only (currently affects personal/admin settings, too)

CC: @nextcloud/designers @MorrisJobke

### CODE REVIEW

**Comment by raghunayyar** [comment]
Personally, I find it a bit too noisy now since the left ones have the heading on the left and the centre ones have the heading below the image. There is no visual consistency here. May be you want to use two of them instead of the three and make sure the image comes on the left and the content on the right.

**Comment by raghunayyar** [comment]
Aaah, I missed the second image. Never mind. :P

**Comment by tcitworld** [comment]
I think margin for only two columns would be fine.

**Comment by eppfel** [comment]
I like the idea, but it could really be a bit too "noisy". The styling of the per app layout is not so easy, because these are inline-box elements right now.
Experimental apps are separated now:
bildschirmfoto 2016-09-10 um 17 06 16
bildschirmfoto 2016-09-10 um 17 06 30

**Comment by rullzer** [comment]
Thanks for the feedback. I just wanted to get things started. As I have the artistic insight of a table I'll happily hand this over :) I just found that the current design had way to much wasted space.

@eppfel thnx!

**Comment by eppfel** [comment]
My try in making it less noisy:
bildschirmfoto 2016-09-11 um 11 49 33
bildschirmfoto 2016-09-11 um 11 51 57

I think the elements of each app have to be rearranged for a proper overall look. Maybe we have to take the design of the app store into account as well: https://apps.nextcloud.com/

**Comment by jancborchardt** [comment]
@eppfel looks awesome! (Did I ask you already if you’re going to come to the [Nextcloud conference](https://conf.nextcloud.com)?) ;)

**Comment by eppfel** [comment]
@jancborchardt You did and I will 😉

**Comment by jancborchardt** [comment]
@eppfel @rullzer is a first version of this ready for getting in, or where do you need help? We should merge it to master soon so we can see how it works and polish it for Nextcloud 11. :)

File: settings/css/settings.css [hunk 1]
```diff
@@ -383,12 +383,24 @@ span.version {
...
+	flex-wrap: wrap;
+	align-content: flex-start;
 }
 .section {
 	position: relative;
+	flex: 1 0 330px;
+    margin: 0;
+    border-color: #eee;
+    border-width: 0 1px 1px 0;
+    border-style: solid;
```
Comment (MorrisJobke) [inline settings/css/settings.css]: Tabs vs spaces ;)

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

**Comment by eppfel** [comment]
@jancborchardt It needs to be fixed, because right now it affects the personal settings page as well. Besides that, the flexing seems robust, but the current version still has borders and I would like to get another opinion on that besides yours ;) Maybe there is time during the meetup, but otherwise the week is packed tight.

File: settings/css/settings.css (hunk 1 above)
Comment (eppfel) [inline settings/css/settings.css]: sry, fixed

**Comment by MorrisJobke** [comment]
@eppfel Ready to review? Then change the label ;)

**Review by MorrisJobke — DISMISSED** [review]

**Comment by eppfel** [comment]
@MorrisJobke Not, yet. I just fixed the spaces with an evil force push. But pssst :speak_no_evil: don't tell anyone

**Comment by jancborchardt** [comment]
@MorrisJobke do you have any take on grid lines vs no grid lines? As @eppfel asked above ;)

**Comment by MorrisJobke** [comment]
> @MorrisJobke do you have any take on grid lines vs no grid lines? As @eppfel asked above ;)

I would either go with a higher padding of 40px and no grid lines or leave the grid lines and 30px padding. The higher padding is needed to compensate for the missing visual guidelines. Generally speaking I like the no-gridlines version more, because we only have the line for the sidebar and horizontal lines in the files app for example, but never used vertical grid lines.

**Comment by jancborchardt** [comment]
@eppfel you wanna try out the higher padding? Then let’s get this in and refine later. I’d hate to have this great improvement laying around for too long. ;)

**Comment by eppfel** [comment]
Jep. I'll try that out. Was just a crazy last week, no time at all.

**Comment by eppfel** [comment]
I only used the padding-right, but 50px, because this side has no sharp edge. Works for me, but I will open another issue for organizing the elements of each app.
I squashed my commits, so ready for gogo!

**Comment by LukasReschke** [comment]
LGTM

**Comment by MorrisJobke** [comment]
Tested and works 👍 

Tested in Firefox, Chrome, Safari, IE11 and Edge

**Comment by jancborchardt** [comment]
Awesome work @eppfel! :) :rocket:

**Pull request merged**